### PR TITLE
Fixes #5664

### LINKED ISSUE #5664: [0.10.0] panic in model.Points.scanTo
I'm using `telegraf` feeding into `influxdb`. In order to compute the `% CPU` usage of a container as a time series, it is necessary to compute the _change_ in measurements like `docker_cpu.total_usage`. Since expressions of aggregates and the function `difference()` aren't supported yet, I thought of using a continuous query to precompute these deltas. Basically, within a `GROUP BY` time interval, compute `LAST(total_usage) - FIRST(total_usage)` (or `MAX(total_usage) - MIN(total_usage)`, they should be the same).

What I'm running into is a panic and `influxdb` crashes.

* Is my query messed up? Or,
* Is this a bug in `influxdb`?

I've included the backtrace and the query in [this gist](https://gist.github.com/dvusboy/96e7e98bd2aac5e1d875).

### PR DESCRIPTION
**Pull request by e-dard**
This fixes the panic in #5664.

### CODE REVIEW

**Comment by jwilder** [comment]
:+1:

**Comment by jonseymour** [comment]
@e-dard See the test case in #5712. I have actually encountered this in a live system. I was getting a panic prior to apply a version of #5672. After I applied #5672 the system went into an infinite loop. 

Reading the code, I discovered newFieldsFromBinary doesn't handle the empty field name case.

I am going to dig into why my system is trying to use a field with an empty field name.

**Comment by jonseymour** [comment]
Ok, I found the problem that is resulting in an empty field name which is then generating the panic &/or loop.

If I run this query:

select count(value) into "summary" from "foobar" where time > now() - 1h group by time(5m)

everything works as expected - a new series is created which has a "count" field.

However, if I do this:

select count(value)**/60** into "summary" from "foobar" where time > now() - 1h group by time(5m)

then I trigger either the panic or the infinite loop depending on which version of the code I have installed.

In other words, the empty field name results from using math in the select clause of a SELECT INTO statement.

**Comment by jonseymour** [comment]
Ok, I combined the test in #5712 with a fix for the problem demonstrated by that test into #5716 and have withdrawn #5712.

There is a related PR #5714 which fixes a problem that causes binary points to be created with empty field names.

Both PRs currently merge cleanly with 0.10.0. #5714 will need a merge conflict to be resolved before it is ready for master.

**Comment by jonseymour** [comment]
This PR could be closed provided #5716 is accepted, since #5716 also includes this change. See 88937ab0

**Comment by e-dard** [comment]
Closed in favour of #5716
